kairon/shared/concurrency/actors/utils.py

```
@@ -0,0 +1,173 @@
+            else:
+                value = item.get('value')
+
+            if query_type == "payload_search":
+                try:
+                    if isinstance(value, str):
+                        request_payload["payload_search"] = json.loads(value)
+                    else:
+                        request_payload["payload_search"] = value
+                except json.JSONDecodeError as e:
+                    logger.debug(e)
+                    raise Exception(f"Error converting payload to JSON: {value}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)